src/pepr/operator/controllers/keycloak/client-secret-sync.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { K8s, kind } from "pepr";
+import { v4 as uuidv4 } from "uuid";
+import { Component, setupLogger } from "../../../logger";
+
+export const KEYCLOAK_CLIENT_SECRET_KEY = "uds-operator";
+
+export const KEYCLOAK_CLIENTS_SECRET_NAMESPACE = "keycloak";
+export const KEYCLOAK_CLIENTS_SECRET_NAME = "keycloak-client-secrets";
+
+const log = setupLogger(Component.OPERATOR_CONFIG);
+
+export async function updateKeycloakClientsSecret(
+  config: kind.Secret,
+  forceRotation: boolean = false,
+) {
+  config.data = config.data || {};
+
+  // This might be a bug but it seems Zarf adds managedFields, which is prohibited in Secrets.
+  delete config.metadata?.managedFields;
```

Comment:
This is actually "normal" and your solve here should work, we've done this with some other updates like our netpols: https://github.com/defenseunicorns/uds-core/blob/c1703221b85d37451d60c226863a0b168e702e01/src/pepr/operator/controllers/network/generators/kubeNodes.ts#L128-L131